### PR TITLE
DATAGRAPH-1101: Documentation improvements

### PR DESCRIPTION
**Pull request by meistermeier**
Clean up the documentation.

### CODE REVIEW

**Comment by meistermeier** [comment]
This one is for you @michael-simons https://github.com/spring-projects/spring-data-neo4j/pull/427/files#diff-923dea2a37ec85849fdde2d608f250a7R325

File: src/main/asciidoc/reference/getting-started.adoc [hunk 1]
```diff
@@ -322,6 +322,26 @@ The `SessionFactory` is needed by SDN to create instances of `org.neo4j.ogm.sess
...
+    basePackages = "org.neo4j.example.repository",
+    sessionFactoryRef = "mySpecialSessionFactory" // <1>
+)
+@EnableTransactionManagement
+public class MyConfiguration {
+}
+----
+<1> Use the name of the session factory you want to be bound to the repositories for that package
+
+In cases as this SDN also supports multiple declarations of `@EnableNeo4jRepositories` that target different packages.
```
Comment (meistermeier) [inline src/main/asciidoc/reference/getting-started.adoc]: cases `like` this.. sounds better to me.

File: src/main/asciidoc/reference/getting-started.adoc [hunk 2]
```diff
@@ -322,6 +322,26 @@ The `SessionFactory` is needed by SDN to create instances of `org.neo4j.ogm.sess
...
+Our recommendation however is connecting one application to one instance.
+If your domains are so different that they are stored in different databases, a microservices approach might be a good fit to your application.
```
Comment (meistermeier) [inline src/main/asciidoc/reference/getting-started.adoc]: `good` -> `better` ?

**Comment by michael-simons** [comment]
Good remarks, thanks. Fixed.

File: src/main/asciidoc/preface.adoc [hunk 3]
```diff
@@ -36,7 +36,7 @@ There are a lot of articles, blog entries and books on the matter - take a look
 == NoSQL and Graph databases
 
 A graph database is a storage engine that is specialised in storing and retrieving vast networks of information.
-It efficiently stores data as nodes and relationships and allows high performance retrieval and querying of those structures.
+It efficiently stores data as nodes with relationships to other nodes and allows high performance retrieval and querying of those structures.
```
Comment (meistermeier) [inline src/main/asciidoc/preface.adoc]: This is nitpicking: A relationship does not necessarily point to _other_ nodes.

File: src/main/asciidoc/reference/neo4j-repositories.adoc [hunk 4]
```diff
@@ -371,7 +371,8 @@ public interface PersonRepository extends Neo4jRepository<Person, Long> {
 === Mapping Query Results
 
 For queries executed via `@Query` repository methods, it's possible to specify a conversion of complex query results to POJOs.
-These result objects are then populated with the query result data and can be serialized and sent to a different part of the application, e.g. a frontend-ui.
+These result objects are then populated with the query result data.
+Those POJOs are easier to handle and might be use as Data Transfer Objects (DTOs) as they are not ttached to a `Session` and don't participate in any lifecycle.
```
Comment (meistermeier) [inline src/main/asciidoc/reference/neo4j-repositories.adoc]: 'they are not **_a_**ttached'

File: src/main/asciidoc/reference/ogm-support.adoc [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 To get started, you need only your domain model and the annotations provided by the Neo4j-OGM library.
 You use annotations to mark domain objects to be reflected by nodes and relationships of the graph database.
-Annotations for individual fields allow you to declare how fields should be processed and mapped to a node in a the graph.
+Annotations for individual fields allow you to declare how fields should be processed and mapped to a nodes properties in the graph.
```
Comment (meistermeier) [inline src/main/asciidoc/reference/ogm-support.adoc]: '...mapped to ~_a_~ nodes properties...'

File: src/main/asciidoc/reference/ogm-support.adoc [hunk 6]
```diff
@@ -58,12 +58,12 @@ In general, the scope of a `Session` should correspond to a "unit of work" in yo
...
-Basic operations are now entirely limited to CRUD operations on entities and executing arbitrary Cypher queries; more low-level manipulation of the graph database is not possible.
+Basic operations are limited to CRUD operations on entities and executing arbitrary Cypher queries; more low-level manipulation of the graph database is not possible.
 
-Given that the latest version of the framework is driven by Cypher queries alone, there's no way to work directly with `Node` and `Relationship` objects any more in remote server mode.
+Given that the latest version of the framework is driven by Cypher queries alone, there's no way to work directly with `Node` and `Relationship` objects, especially in remote server mode.
 Similarly, the `traverse()` method has disappeared, again because the underlying query-driven model doesn't handle it in an efficient way.
 
-If you find yourself in trouble because of the omission of these features, then your best options are:
+If you did rely on low-level manipulation of the graph database itself or being able to use `travers()` in previous versions of Spring Data Neo4j, then your best options are:
```
Comment (meistermeier) [inline src/main/asciidoc/reference/ogm-support.adoc]: ..`travers**_e_**()`

**Comment by michael-simons** [comment]
Closed with 05d045f86e4867df3b3e80fb05cd09715aae44a5.

**Pull request closed without merging**